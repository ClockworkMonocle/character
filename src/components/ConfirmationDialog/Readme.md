Used to show contextual information to a user with discrete choices. ConfirmationDialogs usually have titles, content, and buttons.

```js
require('../Button');
require('../Type');
const toggle = () => setState({ open: !state.open });

<div>
  <Button onClick={toggle}>Show Dialog</Button>
  <ConfirmationDialog
    onDismiss={toggle}
    active={state.open}
    title={<Type color="white" variant="heading-1">Check out this stuff</Type>}
    body={<Type>Here is some really cool things that you might need to check out. Or even edit. Maybe you need to edit a bunch of stuff while in this dialog.</Type>}
    controls={[
      <Button onClick={toggle} key="btn1" variant="bare" color="green">Ok</Button>,
      <Button onClick={toggle} key="btn2" variant="bare" color="red">Cancel</Button>,
    ]}
  />
</div>
```
