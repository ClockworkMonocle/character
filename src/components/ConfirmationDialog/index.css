
.confirmation_dialog {
  background-color: var(--white);
  border-radius: 2px;
  max-width: 500px;
  max-height: 85vh;
}

.confirmation_dialog__title {
  background-color: var(--dark);
  padding: 12px;
}

.confirmation_dialog__body {
  padding: 12px;
}

.confirmation_dialog__controls {
  padding: 12px;
  display: flex;
  justify-content: flex-end;
}

.confirmation_dialog__controls > *:first-child {
  margin-right: 24px;
}
