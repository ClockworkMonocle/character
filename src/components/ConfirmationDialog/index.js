import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'components/Modal';

import './index.css';

export default function ConfirmationDialog({ active, title, body, controls, onDismiss }) {
  return (
    <Modal active={active} onDismiss={onDismiss}>
      <div className="confirmation_dialog">
        <div className="confirmation_dialog__title">
          {title}
        </div>
        <div className="confirmation_dialog__body">
          {body}
        </div>
        <div className="confirmation_dialog__controls">
          {controls}
        </div>
      </div>
    </Modal>
  );
}

ConfirmationDialog.propTypes = {
  active: PropTypes.bool,
  title: PropTypes.node.isRequired,
  body: PropTypes.node.isRequired,
  controls: PropTypes.node.isRequired,
  onDismiss: PropTypes.func.isRequired,
};
