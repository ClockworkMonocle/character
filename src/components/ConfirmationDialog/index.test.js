import React from 'react';
import { shallow } from 'enzyme';
import ConfirmationDialog from './index';

test('<ConfirmationDialog /> renders', () => {
  expect(shallow(<ConfirmationDialog
    title="test-dialog"
    body={<p>yo</p>}
    controls={[
      <p key="a">a</p>,
      <p key="b">b</p>,
    ]}
    onDismiss={() => {}}
  />)).toMatchSnapshot();
});
