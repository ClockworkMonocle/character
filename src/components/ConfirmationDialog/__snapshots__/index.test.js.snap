// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationDialog /> renders 1`] = `
<Modal
  active={false}
  from="middle"
  onDismiss={[Function]}
>
  <div
    className="confirmation_dialog"
  >
    <div
      className="confirmation_dialog__title"
    >
      test-dialog
    </div>
    <div
      className="confirmation_dialog__body"
    >
      <p>
        yo
      </p>
    </div>
    <div
      className="confirmation_dialog__controls"
    >
      <p
        key="a"
      >
        a
      </p>
      <p
        key="b"
      >
        b
      </p>
    </div>
  </div>
</Modal>
`;
